### ISSUE TITLE
[tune] search space API

### ISSUE THREAD

**Issue by sumanthratna** [ray-project/ray#9969]
### Describe your feature request
I'm working on a project where we want the user to be able to choose which algorithm they want to use via the command line. This makes for ugly code—for example, see:
https://github.com/MLTransfer/MLEncrypt-Research/blob/ce2c6b5fbf77d2486ec4a702a4bdf63dcdba2330/mlencrypt_research/cli.py#L446-L635

Dedicating almost 200 lines of code to instantiating the search algorithm seems unnatural, and this same problem occurs for choosing schedulers. 

At least for search algorithms, this is mainly because most algos only work with 1 hparam configuration format. I think a reasonable way to deal with this is to force the user to stick to a single format, and then tune converts this to the appropriate format. Using ConfigSpace seems reasonable since it's meant for hparam configs and is already required by BOHB.

One problem with implementing this is that different search algorithms accept different arguments. An easy way to work around this is to just accept `**kwargs` in the shim and pass these to the initializer of the actual search algorithm object. 

TL;DR something that looks like this:
```python
hparam_config = ConfigSpace()  # properly instantiate this
search_alg = ray.tune.suggest.create('dragonfly', hparam_config, **kwargs)
scheduler = ray.tune.schedulers.create('asha', **kwargs)
tune.run(trainable, search_alg=search_alg, scheduler=scheduler)
```

---

Originally suggested to @richardliaw in his Slack DMs.

---

edit: see #10451 for shim instantiation

**Comment by richardliaw** [comment]
yes!

**Comment by richardliaw** [comment]
I'll get in touch with you next week with a proposal.

**Comment by sumanthratna** [comment]
@richardliaw we can close this right? #10444 and #10518 have been merged into master

**Comment by richardliaw** [comment]
yep! sounds good. Does your code become much cleaner now?

**Comment by sumanthratna** [comment]
> Does your code become much cleaner now?

wayyyy cleaner :) thanks for making this a priority! can't wait for 1.0
